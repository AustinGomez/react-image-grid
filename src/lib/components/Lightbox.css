.lightbox-window {
    position: fixed;
    top: 50%;
    left: 50%;
    height: 100%;
    width: 100%;
    -webkit-transform: translate(-50%,-50%);
    -moz-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    -o-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    z-index: 11;
    background: black;
    display: flex;
}

.lightbox-header-shadow {
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    height: 72px;
    background-image: linear-gradient(rgba(0, 0, 0, 0.288), rgba(0, 0, 0, 0))
}

.lightbox-image-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    margin: auto;
}

#lightbox-image {
    max-width: 100%;
    max-height: 100%;
}

.back-button {
    height: 20px;
    position: absolute;
    left: 25px;
    top: 30px;
    cursor: pointer;
}

.arrow-container {
    height: 80%;
    width: 30%;
    cursor: pointer;
    display: flex;
    align-items: center;
    position:absolute;
    cursor: pointer;
    -webkit-tap-highlight-color: transparent;
}

/* override for the right arrow only */
.arrow-container.right-arrow {
    justify-content: flex-end;
}

.left-arrow {
    left: 30px;
}

.right-arrow {
    right: 30px;
}

.arrow-container img {
    height: 72px;
}



.lightbox-backdrop {
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    z-index: 10;
    background-color: rgba(0, 0, 0, 0.5);
}
