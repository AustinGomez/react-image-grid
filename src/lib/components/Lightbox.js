import React, { useState, useEffect } from 'react'
import './Lightbox.css'

const Lightbox = ({ selectedImage, onNext, onPrev, onClose }) => {
    const [showLeftArrow, setShowLeftArrow] = useState(false)
    const [showRightArrow, setShowRightArrow] = useState(false)

    const handleKeyPress = (e) => {
        switch (e.code) {
            case 'Escape':
                onClose()
                break
            case 'ArrowRight':
                onNext()
                break
            case 'ArrowLeft':
                onPrev()
                break
            default:
                break
        }
    }

    useEffect(() => {
        window.addEventListener('keyup', handleKeyPress)
        return (() => {
            window.removeEventListener('keyup', handleKeyPress)
        })
    }, [])

    return (
        <>
            <div className="lightbox-backdrop" onClick={onClose}></div>
            <div className="lightbox-window">
                <div className="lightbox-header-shadow">
                    <img className="back-button" src="icons/left-arrow.svg" onClick={onClose} />
                </div>
                <div className="lightbox-image-container">
                    <div
                        className="arrow-container left-arrow"
                        onClick={onPrev}
                        onMouseEnter={() => setShowLeftArrow(true)}
                        onMouseLeave={() => setShowLeftArrow(false)}>
                        {showLeftArrow ? <img src="icons/chevron-left.svg" /> : null}
                    </div>
                    <img id="lightbox-image" src={selectedImage.download_url} alt={selectedImage.alt} />
                    <div className="arrow-container right-arrow"
                        onClick={onNext}
                        onMouseEnter={() => setShowRightArrow(true)}
                        onMouseLeave={() => setShowRightArrow(false)}>
                        {showRightArrow ? <img src="icons/chevron-right.svg" /> : null}
                    </div>
                </div>
            </div>
        </>
    )
}

export default Lightbox