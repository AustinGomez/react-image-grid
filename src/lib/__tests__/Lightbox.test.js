import Lightbox from "../components/Lightbox";
import React from "react";
import { shallow, mount, render } from "enzyme";

describe("Lightbox component", () => {
  let images;
  let wrapper;
  let onNextSpy, onPrevSpy, onCloseSpy;
  beforeEach(() => {
    onNextSpy = jest.fn();
    onPrevSpy = jest.fn();
    onCloseSpy = jest.fn();
    wrapper = shallow(
      <Lightbox
        selectedImage={"fake"}
        onClose={onCloseSpy}
        onPrev={onPrevSpy}
        onNext={onNextSpy}
      />
    );
  });
  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call the prev handler on click of left button", () => {
    wrapper.find(".left-arrow").simulate("mouseEnter");
    wrapper.find(".left-arrow").simulate("click");
    expect(onPrevSpy).toHaveBeenCalledTimes(1);
  });

  it("should call the prev handler on keypress of ArrowLeft", () => {
    const map = {};
    window.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
    const wrapper = mount(
      <Lightbox
        selectedImage={"fake"}
        onClose={onCloseSpy}
        onPrev={onPrevSpy}
        onNext={onNextSpy}
      />
    );
    map.keyup({ code: "ArrowLeft" });
    expect(onPrevSpy).toHaveBeenCalledTimes(1);
  });

  it("should call the next handler on click of right button", () => {
    wrapper.find(".right-arrow").simulate("mouseEnter");
    wrapper.find(".right-arrow").simulate("click");
    expect(onNextSpy).toHaveBeenCalledTimes(1);
  });

  it("should call the next handler on keypress of ArrowRight", () => {
    const map = {};
    window.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
    const wrapper = mount(
      <Lightbox
        selectedImage={"fake"}
        onClose={onCloseSpy}
        onPrev={onPrevSpy}
        onNext={onNextSpy}
      />
    );
    map.keyup({ code: "ArrowRight" });
    expect(onNextSpy).toHaveBeenCalledTimes(1);
  });

  it("should call the close handler on click of close button", () => {
    wrapper.find(".back-button").simulate("click");
    expect(onCloseSpy).toHaveBeenCalledTimes(1);
  });

  it("should call the close handler on keypress of Escape", () => {
    const map = {};
    window.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
    const wrapper = mount(
      <Lightbox
        selectedImage={"fake"}
        onClose={onCloseSpy}
        onPrev={onPrevSpy}
        onNext={onNextSpy}
      />
    );
    map.keyup({ code: "Escape" });
    expect(onCloseSpy).toHaveBeenCalledTimes(1);
  });
});
