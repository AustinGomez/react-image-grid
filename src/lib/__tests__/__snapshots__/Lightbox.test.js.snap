// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightbox component should render without crashing 1`] = `
<Fragment>
  <div
    className="lightbox-backdrop"
    onClick={[MockFunction]}
  />
  <div
    className="lightbox-window"
  >
    <div
      className="lightbox-header-shadow"
    >
      <img
        alt="back-button-icon"
        className="back-button"
        onClick={[MockFunction]}
        src="icons/left-arrow.svg"
      />
    </div>
    <div
      className="lightbox-image-container"
      onMouseDown={[Function]}
    >
      <div
        className="arrow-container left-arrow"
        onClick={[MockFunction]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <img
        id="lightbox-image"
      />
      <div
        className="arrow-container right-arrow"
        onClick={[MockFunction]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
